Docs/sdk/control-library-nuget-package-spec.md

```diff
@@ -0,0 +1,79 @@
+# Control library NuGet Package Spec
+
+Windows Forms has a rich ecosystem of vendors that provide controls and other UI components the developers can use to quickly build a fully functioning app.  
+
+In the past (with .NET Framework Windows Forms), control vendors could depend on their designer being loaded into the same process as Visual Studio (VS). This enabled them to call Visual Sudio API and manipulate the painting of their controls directly from within VS.  
+
+However, .NET Core Windows Forms is rendered in a separate process from VS, so control authors are going to need a mechanism to package their controls that works with this new architecture.  
+
+NuGet is the most popular way to package libraries in .NET, and vendors tend to ship their proprietary libraries by installing them to a local folder as NuGet packages, and turning that into a NuGet feed for the developer.  
+
+In order to make this scenario work today, control library developers need to be able to specify which assemblies should be loaded by Visual Studio, and which assemblies should be loaded by the Windows Forms Design Tools Server process. This will be done by a folder convention within the NuGet package itself. 
+
+## Prerequisites
+
+Your package must build against the [WinForms Designer SDK](https://www.nuget.org/packages/Microsoft.WinForms.Designer.SDK).
+
+## Assembly Locations
+
+Windows Forms will attempt to load from the following locations: 
+
+1. Runtime controls are expected to be in `lib\<tfm>` 
+   * `tfm` stands for Target Framework Moniker (`net472`, `netcoreapp3.1`, or `net5.0`, for example) 
+   * This should be the framework that the runtime controls target 
+1. Visual Studio designers are expected to be in `lib\<tfm>\Design\WinForms` 
+   * These should all target `.NET Framework 4.7.2`, since that's the framework that Visual Studio targets 
+1. WinForms Design Tools Server process designers are expected to be in `lib\<tfm>\Design\WinForms\Server` 
+   * These should all target `.NET Core 3.1` (or whatever framework the WinForms Design Tools Server process is targeting) 
+1. There is no specific naming convention for the assemblies - WinForms attempts to load them all
+
+If `lib\<tfm>\Design\WinForms` exists, we will only load files under that root. If it doesn’t, we will fall back to using the `lib\<tfm>\Design` folder as the root.  
+  * This allows control developers to ship Windows Forms and WPF controls and designers in the same package (without file collisions) if they wish.
+  
+## Scenarios and User Experience
+
+### Scenario 1: A control with no custom designer
+
+For a runtime control that does not have a custom designer (`ProgressBar`, for example) we would simply use the default designer. The control files go under `lib\<tfm>`:
+
+  ![](/docs/sdk/images/1.png)
```

Comment:
I would prefer descriptive names for png files. And please put a good description into the `[]` to be used as an Alt-text

Reply:
Absolutely right, great catch. I've cleaned it up.

Reply:
Did you push this change?

Reply:
I just did, good catch.